Mathlib/Analysis/Analytic/Meromorphic.lean

```diff
@@ -310,6 +339,50 @@ lemma id {U : Set 𝕜} : MeromorphicOn id U := fun x _ ↦ .id x
 lemma const (e : E) {U : Set 𝕜} : MeromorphicOn (fun _ ↦ e) U :=
   fun x _ ↦ .const e x
 
+/-- The set where a meromorphic function has infinite order is clopen in its domain of meromorphy.
+-/
+theorem clopen_of_order_eq_top {U : Set 𝕜} [CompleteSpace E] (h₁f : MeromorphicOn f U) :
+    IsClopen { u : U | (h₁f u.1 u.2).order = ⊤ } := by
+  constructor
+  · rw [← isOpen_compl_iff, isOpen_iff_forall_mem_open]
+    intro z hz
+    rcases (h₁f z.1 z.2).eventually_eq_zero_or_eventually_ne_zero with h | h
+    · -- Case: f is locally zero in a punctured neighborhood of z
+      rw [← (h₁f z.1 z.2).order_eq_top_iff] at h
+      tauto
+    · -- Case: f is locally nonzero in a punctured neighborhood of z
+      obtain ⟨t', h₁t', h₂t', h₃t'⟩ := eventually_nhds_iff.1 (eventually_nhdsWithin_iff.1
+        (h.and (h₁f z.1 z.2).eventually_analyticAt))
+      use Subtype.val ⁻¹' t'
+      constructor
+      · intro w hw
+        simp
+        by_cases h₁w : w = z
+        · rwa [h₁w]
+        · have h₂f := (h₁t' w hw) (Subtype.coe_ne_coe.mpr h₁w)
+          simp [h₂f.2.meromorphicAt_order, h₂f.2.order_eq_zero_iff.2 h₂f.1]
+      · exact ⟨isOpen_induced h₂t', h₃t'⟩
+  · apply isOpen_iff_forall_mem_open.mpr
+    intro z hz
+    simp_rw [MeromorphicAt.order_eq_top_iff, eventually_nhdsWithin_iff, eventually_nhds_iff] at *
+    obtain ⟨t', h₁t', h₂t', h₃t'⟩ := hz
+    use Subtype.val ⁻¹' t'
+    simp [isOpen_induced h₂t', h₃t']
```

Comment:
You should avoid nonterminal simps, i.e., bare `simp` like you are doing here, in the middle of the proof: if someone changes the simpset (the set of lemmas that simp appplies) in another file, then your proof could break because the state after the `simp` would change. Instead, use `simp only`, with the right set of lemmas, so that it is more robust. To do that, use `simp?` to see which lemmas your simp call uses.

Same thing below.

Reply:
Done.